### ISSUE TITLE
Bug: Macros aren't executed properly

### ISSUE THREAD

**Issue by chrisgrieser** [ja-ford/delaytrain.nvim#8]
When *executing* a macro and one of the delay-train letters occurs during the macro, the execution is somehow messed up. Seems to be a regression of #1 or something.

my config:
```lua
require("delaytrain").setup {
	delay_ms = 1000, 
	grace_period = 7,
	keys = { 
		["nv"] = { "h", "l" },
	},
}
```

- macro I recorded: `ralral`
- resulting effect when executing the macro: `rall`

**Comment by ja-ford** [comment]
Yeah this is definitely an issue with how we're sending the keypress. Even if I disable DelayTrain before recording the macro, execution gets tripped up whenever chained keys are used (such as `r`, `d`, `s`, etc).

My guess is that neovim is still running DelayTrain's `try_delay_keypress` function while the rest of the macro is executing, so the delayed keypresses are being sent late due to the overhead, even though it just sends the keypress while macros are executing. I think we would have to figure out a different way to ignore on macro execution, such as disabling DelayTrain when execution starts. Unfortunately there's no autocmd like "MacroExecutionEnter" so I'm not sure what the better option is, but I'll keep an eye on this and try to see what our options are.

**Comment by chrisgrieser** [comment]
Maybe there is some way to add `reg_executing` as a condition somewhere? https://neovim.io/doc/user/builtin.html#reg_executing()

**Comment by ja-ford** [comment]
That's how the code currently works. When a keypress is attempted, we check `reg_executing` to see if a macro is being run, and default to always send a keypress is true. I think the problem is that there's still some extra time being spent in DelayTrain's code checking for macros and forwarding the keypress, so subsequent keypresses in the macro may be getting sent first.

For example, I tried this with a recorded macro `ralrel`, so `string` should become `aering`. But we're seeing `etring` with the cursor on `r`. So it seems like the first `l` is still being processed when `re` is run, causing the cursor not to move before replacing with "e". Afterwards, both `l`s are processed and the cursor is moved two spaces to the right. That's just a theory but I'm not sure what else would cause this.

This may also have more to do with now `nvim_feedkeys` works with macros, so if there's a possible workaround in that area then we may be able to fix this without having to disable DelayTrain before macro execution. Either way there's some research still needed here.

**Comment by chrisgrieser** [comment]
hmmm, for what it's worth, I just stumbled upon [this on the neovim reddit](https://www.reddit.com/r/neovim/comments/103n2gr/code_gist_avoid_repeating_hjkl_lua_way/), and using that snippet, the issue with the macros does not occur anymore. Weird.
